dist/tools/zsh-completion/zsh-riot.sh

```diff
@@ -31,6 +31,48 @@ function _bools {
     _describe 'bool' _bool_vals
 }
 
+function _programmers {
+    local -a _programmer_vals=(
+        "openocd:use OpenOCD for programming via JTAG/SWD (default for most boards)"
+        "jlink:use Segger's JLinkExe for programming (requires original Segger programmers)"
+        "esptool:program ESP32 (S2,S3,C2,C3,C6,...) ESP8266 via UART"
+        "avrdude:default for AVR boards"
+        "lpc2k_pgm:default for lpc23xx boards"
+        "bossa:for SAM based Arduino boards"
+        "edbg:for SAM based evaluation board"
+        "nrfutil:for nRF5x boards with nRF bootloader"
+        "stm32flash:for STM32 boards via UART bootloader"
+        "uniflash:for CC13xx / CC26xx boards"
+        "cc2538-bsl:for CC2538 boards"
+        "mspdebug:for MSP430 boards"
+        "goodfet:for some MSP430 boards"
+        "uf2conf:program via UF2 bootloader (except RP2040)"
+        "elf2uf2:program via UF2 bootloader (RP2040)"
+        "cpy3remed:program via ST-Link using filesystem disk interface"
+        "adafruit-nrfutil:program via Adafruits nRF5x bootloader"
+    )
```

Comment:
`rg PROGRAMMERS_SUPPORTED boards/ | cut -d '='  -f2 | tr ' ' '\n' | sort | uniq` finds some more:

```suggestion
        "bmp:Black Magic Probe",
        "dfu-util:Flash using DFU Bootloader"
        "pyocd:Python based tool and API for debugging, programming, and exploring Arm Cortex microcontrollers"
        "robotis-loader:Flash using ROBOTIS Bootloader"
        "stm32loader:Flash using STM32 Bootloader"
    )
```